### ISSUE TITLE
Project submission / solution naming

### ISSUE THREAD

**Issue by KevinMulhern** [TheOdinProject/theodinproject#3955]
We're using both project submission and project solutions interchangeably throughout the codebase to describe user projects. 

Such as here lol:
<img width="766" alt="Screenshot 2023-07-11 at 21 29 38" src="https://github.com/TheOdinProject/theodinproject/assets/7963776/585392e0-4a6d-4386-a480-1fc479423ad1">

This isn't a huge problem, but it's always a good idea to be consistent. So which should it be?

**Comment by ChargrilledChook** [comment]
I don't feel super strongly about this either way, but I think it would be good to be consistent. A few thoughts:

- Submission seems to be the prevailing name in code. There's a project submission model, component and a few other things. Making it submission would be a lower friction way to be consistent
- User facing side doesn't have to match up to the code but I think matching code and domain language is preferable
- Solution is ok for smaller projects, but feels a bit odd for some of the larger projects. Makes it seem more like homework than a potential portfolio piece, for instance

For those reasons I'd go with submission if we wanted to be consistent, but not a hill I'd die on

**Comment by KevinMulhern** [comment]
Thanks @ChargrilledChook, great points 👏 I've been giving this way more thought than it really deserves the past couple of days 😂

I've been landing on solution probably being the more accurate domain name - _"a user submits their **solution** for a project"_ kind of thing.

Tracing things back, I think submission came from when users had to submitt PRs to add their projects. We called them project submissions internally within the team and that terminology found its way into the feature. But learners almost always refer to them as solutions based on some Discord searching.

I also had a look at what other platforms are calling this kind of thing. [Codementor](https://www.codementor.io/projects) is probably the closest to our project feature, but I looked at FCC, Exercism, Codewars and others too. "Solution" is prevalent across them all.

Although, I do agree **a lot** with what you're saying about it feeling odd for larger projects. I wish there was something better, something _cooler_ lol we could use. And you're not wrong about the PR to change it being a pain, it's going to be big 😬

**Comment by ManonLef** [comment]
I agree with Chook. Especially that I wouldn't consider bigger projects a "solution". Between those two, I'd vote "submission"

Alternatives:
- finished projects
- projects
- creations (like creations showcase)
- submitted projects
- subbos (to throw in a `cool` one that should absolutely be disregarded as a serious suggestion. context:)

>"I wish there was something better, something cooler lol we could use" -*Kevin*

**Comment by Asartea** [comment]
While I agree that between solutions and submissions is the better, I still think submissions seems to suggest that you are like submitting it for review? In that way I could see just a plain `project` as main name working, with either finished or submitted as prefixes depending on the situation (so: "when you are done you can submit your finished project" but eg: "Showing 10 most liked projects", "View full list of submitted projects")

**Comment by KevinMulhern** [comment]
I wish we could use plain project/s. But within the overall system, project is as good as a reserved word - it's the project itself.

> I still think submissions seems to suggest that you are like submitting it for review?

Thats my sums up my problem with submissions - it implies it's a proposal that needs some action like a review or judgement after it's been submitted.

I do agree solutions doesn't accurately describe bigger projects for the learner outside of TOP. Hopefully, for them they are portfolio pieces.  But it is pretty accurate within the context of a project page. Any items in the list are community solutions for the project. 

Having spent _way_ too much time obsessing over what to name things in my career lol - one of the things I've learnt is in the absence of a completely bullet proof name. The next best thing is going with what most people are already calling it. On Discord at least, thats solutions by a significant margin.

**Comment by thatblindgeye** [comment]
Personally feel like "solutions" is more apt to things like our exercise repos. "Here's my solution to solve this exercise." While submissions feels a little more apt to projects, since it's not really "Here's my solution to this project." Either way there's going to be some unintended connotation to either word. "Well solution implies we're solving a problem when we're just creating something"/"Submission sounds like we're waiting for review". And we can just as easily convey our intent for either word.

I do like Manon's suggestion of just "project". The only possible issue with that is there might be some Ruby "projects" that are more akin to exercises? Things like the [basic ruby projects](https://www.theodinproject.com/paths/full-stack-ruby-on-rails/courses/ruby#basic-ruby-projects). Though that could open another can of worms - should those "projects" actually be part of a Ruby exercise repo instead (not sure if this is something already in the works/discussed so the Team Magma people will have to chime in here).

**Comment by rlmoser99** [comment]
> But within the overall system, project is as good as a reserved word 

From what I understand, using the word `project` is not an option. I think the word `solution` works for a big project like TicTacToe and for a small exercise like the Ruby Basic Projects.

**Comment by Asartea** [comment]
I still think that if we restrict ourselves to just `solution` vs.`submission` submission is by far the best option. At least then you can explain it away by saying you are submitting your project, while solution implies there was a problem to be solved, which just isn't true for (almost) all projects.

**Comment by Ariech** [comment]
I agree with Manon's suggestion regarding the term `project`. I think it perfectly captures what users have been working on. On Discord, most people use the term `solutions` when sharing answers to the assigned exercises. The term `submission` implies that our work will be evaluated. If using the word `project` is not possible, then I would go with `solution` as it accurately conveys what is expected from us - finding a resolution to the given task.

**Comment by JustWaveThings** [comment]
What about 'User Project' for non-exercise submissions (tic-tac-toe) and "User Exercise" for user exercises like the css exercise repo?
